File: IN1010/trix/go/kvadratstabel/kvadratStabel.go
package main

type KvadratStabel struct {
	hode *node
}

type node struct {
	data *Kvadrat
	neste *node
}

func (ks *KvadratStabel) LeggPaa(k *Kvadrat) {
	ny := &node{
		data: k,
		neste: ks.hode,
	}
	ks.hode = ny
}

func (ks *KvadratStabel) ErTom() bool {
	return ks.hode == nil
}

func (ks *KvadratStabel) TaAv() *Kvadrat {
	if ks.ErTom() {
		return nil
	}
	ret := ks.hode
	ks.hode = ks.hode.neste
	return ret.data
}

File: IN1010/trix/06/kvadratstabel/Kvadratstabel.java
class Kvadratstabel {
    Node start;

    class Node {
        Kvadrat data;
        Node neste;

        public Node(Kvadrat data, Node neste) {
            this.data = data;
            this.neste = neste;
        }
    }

    public void leggPaa(Kvadrat k) {
        Node ny = new Node(k, null);
        if (this.start == null) {
            this.start = ny;
            return;
        }
        Node tmp = this.start;
        ny.neste = tmp;
        this.start = ny;
    }

    public Kvadrat taAv() {
        Node ret = this.start;
        if (ret == null) {
            return null;
        }
        this.start = this.start.neste;
        return ret.data;
    }

    public boolean erTom() {
        return this.start == null;
    }
}
